contracts/attribute/Kept/Kept.sol

```diff
@@ -33,22 +33,27 @@ abstract contract Kept is IKept, Initializable {
     /// @param data Arbitrary data passed in from the keep modifier
     function _raiseKeeperFee(UFixed18 amount, bytes memory data) internal virtual { }
 
+    /// @dev Hook for inheriting contracts to perform logic to calculate the dynamic fee
+    /// @param callData The calldata that will be used to price the dynamic fee
+    function _calculateDynamicFee(bytes memory callData) internal view virtual returns (UFixed18) { }
+
     /// @notice Placed on a functon to incentivize keepers to call it
     /// @param multiplier The multiplier to apply to the gas used
     /// @param buffer The fixed gas amount to add to the gas used
     /// @param data Arbitrary data to pass to the _raiseKeeperFee function
-    modifier keep(UFixed18 multiplier, uint256 buffer, bytes memory data) {
+    modifier keep(UFixed18 multiplier, uint256 buffer, bytes memory dynamicCalldata, bytes memory data) {
         uint256 startGas = gasleft();
 
         _;
 
+
         uint256 gasUsed = startGas - gasleft();
         UFixed18 keeperFee = UFixed18Lib.from(gasUsed)
             .mul(multiplier)
             .add(UFixed18Lib.from(buffer))
-            .mul(_etherPrice())
-            .mul(UFixed18.wrap(block.basefee));
-
+            .mul(UFixed18.wrap(block.basefee))
```

Comment:
can you put this back in the same order as before (with just the dynamicCalldata added to the end). The `basefee` line is a divisor since it's < 1, so that should come after the the `_etherPrice` multiple.

Reply:
The dynamic fee is in ETH. Alternatively we get the ETHUSD price and then do the two parts separately

Reply:
💀 We run into a Stack to Deep issue if we save the `_etherPrice` to a separate variable to do individual calculations then add them

```
        UFixed18 ethPrice = _etherPrice();
        UFixed18 keeperFee = UFixed18Lib.from(gasUsed)
            .mul(multiplier)
            .add(UFixed18Lib.from(buffer))
            .mul(ethPrice)
            .mul(UFixed18.wrap(block.basefee))
            // Add the dynamic fee in keeper token
            .add(_calculateDynamicFee(dynamicCalldata).mul(ethPrice));
            ```